### PR TITLE
support CAS identity provider

### PR DESCRIPTION
**Pull request by wansir**
Signed-off-by: hongming <talonwan@yunify.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Support CAS identity provider.

**Which issue(s) this PR fixes**:

Related to  #3047.

**Special notes for reviewers**:
```
```

**Additional documentation, usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by zryfish** [comment]
Please resolve these conflicts.

**Comment by wansir** [comment]
> Please resolve these conflicts.

done

File: pkg/kapis/oauth/handler.go [hunk 1]
```diff
@@ -172,9 +172,13 @@ func (h *handler) Authorize(req *restful.Request, resp *restful.Response) {
 }
 
 func (h *handler) oauthCallback(req *restful.Request, resp *restful.Response) {
-	code := req.QueryParameter("code")
 	provider := req.PathParameter("callback")
-
+	// OAuth2
```
Comment (zryfish) [inline pkg/kapis/oauth/handler.go:176]: comments should be more readable.

File: go.mod [hunk 2]
```diff
@@ -730,7 +732,6 @@ replace (
 	k8s.io/klog/v2 => k8s.io/klog/v2 v2.0.0
 	k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20200410145947-61e04a5be9a6
 	k8s.io/kubectl => k8s.io/kubectl v0.18.6
-	k8s.io/kubernetes => k8s.io/kubernetes v1.14.0
```
Comment (zryfish) [inline go.mod:733]: rebase with master

Comment (wansir) [reply go.mod:733]: Already rebase, https://github.com/kubesphere/kubesphere/blob/master/go.mod#L733

**Comment by zryfish** [comment]
@RolandMa1986 Any comments?

**Pull request merged**